include/constructors/constructors.hpp

```diff
@@ -53,12 +53,7 @@ BigInt::BigInt(const std::string& num) {
     if (num[0] == '+' or num[0] == '-') {     // check for sign
         std::string num_magnitude = num.substr(1);
         if (is_valid_number(num_magnitude)) {
-            /*
-                TODO
-                ----
-                magnitude = convert_to_base_2_to_the_64(num_magnitude);
-            */
-            magnitude = {0};
+            magnitude = convert_to_base_64(num_magnitude);
```

Comment:
It seems we have a circular dependency here, because in order to get magnitudes we first need to create a BigInt from string and then get magnitudes by calling a function which returns a magnitude vector, in my case `BigInt::to_base64`. What are your thoughts on this?